### PR TITLE
Implement StorageSubsystem

### PR DESCRIPTION
**Pull request by FireShredder24**
Adds a Subsystem class that provides on-off control of the VictorSP that controls the 2 conveyance motors and the VictorSP that controls the shooter gate motor.

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/StorageSubsystem.jav [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package frc.robot.subsystems;
```
Comment (jordan-powers) [inline src/main/java/frc/robot/subsystems/StorageSubsystem.jav]: There’s a typo in the name of this file. It should be a ‘.java’ file, not a ‘.jav’ file.

File: src/main/java/frc/robot/subsystems/ShooterSubsystem.java [hunk 2]
```diff
@@ -93,7 +93,7 @@ public ShooterSubsystem() {
     shooterMAXRight.setIdleMode(IdleMode.kCoast);
     shooterMAXLeft.setSmartCurrentLimit(currentLimit);
     shooterMAXRight.setSmartCurrentLimit(currentLimit);
-    // Sets the left shooter motor to follow the right motor, and be inverted.
```
Comment (terbos90803) [inline src/main/java/frc/robot/subsystems/ShooterSubsystem.java]: Try not to mix unrelated cleanup.

File: src/main/java/frc/robot/subsystems/StorageSubsystem.jav (hunk 1 above)
Comment (terbos90803) [reply src/main/java/frc/robot/subsystems/StorageSubsystem.jav]: upvote

File: src/main/java/frc/robot/subsystems/StorageSubsystem.jav [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+public class StorageSubsystem extends SubsystemBase {
+  private final VictorSP StorageGate;
+  private final VictorSP StorageMid;
+
+  public StorageSubsystem() {
+    StorageMid = new VictorSP(Constants.STORAGE_MID_PWM_CHAN);
+    StorageGate = new VictorSP(Constants.STORAGE_GATE_PWM_CHAN);
+  }
+  public runMid() {
+    StorageMid.set(1);
```
Comment (terbos90803) [inline src/main/java/frc/robot/subsystems/StorageSubsystem.jav]: No magic numbers.  Use named constants.

File: src/main/java/frc/robot/subsystems/StorageSubsystem.jav [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  }
+  public runGate() {
+    StorageGate.set(0.1);
+  }
+  public stopMotors() {
+    StorageGate.stopMotors();
```
Comment (terbos90803) [inline src/main/java/frc/robot/subsystems/StorageSubsystem.jav]: You'll likely want to be able to keep the mid running while stopping the gate.  I would recommend an API more along the lines of:
public void runMid(bool run);
public void runGate(bool run);
public void stopAll();

**Review by terbos90803 — CHANGES_REQUESTED** [review]

**Comment by terbos90803** [comment]
Something else important to understand that in FRC, all motor controllers are on watchdog timers.  If you don't feed them a value on every scheduler cycle, they'll safe stop.

This means that if you want to be able to set a state once and forget about it, then you need to change your architecture to hold the desired state in the subsystem class and then set it to the motors in the subsystem's periodic method.

File: src/main/java/frc/robot/subsystems/ShooterSubsystem.java (hunk 2 above)
Comment (FireShredder24) [reply src/main/java/frc/robot/subsystems/ShooterSubsystem.java]: Got it.  For future reference, don't fix unrelated issues you spot while working on a feature branch that are not part of said branch.  Merge conflicts are hell.

File: src/main/java/frc/robot/Constants.java [hunk 5]
```diff
@@ -37,4 +37,8 @@
   // Brushed PWM addresses
   public static final int STORAGE_GATE_PWM_CHAN = 1;
   public static final int STORAGE_MID_PWM_CHAN = 2;
+
+  // Motor value constants
+  public static final double STORAGE_MID_VAL = 1;
+  public static final double STORAGE_GATE_VAL = 0.1;
```
Comment (terbos90803) [inline src/main/java/frc/robot/Constants.java]: Given that these values may need to be tuned, consider making them preferences instead of hardcoded.

File: src/main/java/frc/robot/subsystems/StorageSubsystem.java [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    StorageGate = new VictorSP(Constants.STORAGE_GATE_PWM_CHAN);
+  }
+
+  public void runMid(boolean run) {
+    runMid = run;
+  }
+
+  public void runGate(boolean run) {
+    runGate = run;
+  }
```
Comment (terbos90803) [inline src/main/java/frc/robot/subsystems/StorageSubsystem.java]: How will you handle reversing?

File: src/main/java/frc/robot/subsystems/StorageSubsystem.java [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+    runMid = false;
+    runGate = false;
+  }
+
+  @Override
+  public void periodic() {
+    if (runMid)
+      StorageMid.set(Constants.STORAGE_MID_VAL);
+    if (runGate)
+      StorageGate.set(Constants.STORAGE_GATE_VAL);
```
Comment (terbos90803) [inline src/main/java/frc/robot/subsystems/StorageSubsystem.java]: Actively set the motor on every call.  The value changes depending on the runMid/runGate states.

**Review by terbos90803 — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/utils/MoPrefs.java [hunk 8]
```diff
@@ -6,6 +6,7 @@
   static final double INTAKE_ROLLER_SETPOINT = 0.3;
   static final int CLIMBER_ENCODER_LIMIT = 10000;
   static final double SHOOTER_HOOD_SETPOINT = 15;
+  static final double STORAGE_SPEED_SETPOINT = 1.0;
```
Comment (FireShredder24) [inline src/main/java/frc/robot/utils/MoPrefs.java]: STORAGE_SPEED_SETPOINT seems like a slightly inappropriate name, given that setpoints always mean speed in this context.  I suggest STORAGE_SETPOINT.

Comment (FireShredder24) [inline src/main/java/frc/robot/utils/MoPrefs.java]: STORAGE_SPEED_SETPOINT seems like a slightly inappropriate name, given that setpoints always mean speed in this context. I suggest STORAGE_SETPOINT.

**Review by terbos90803 — APPROVED** [review]
